File: controllers/quantity_pricing_controller.py
from os import getenv
from requests import get


def get_quantity_pricing(catalog_id):
    response = get(
        f"{getenv("SHIFT4SHOP_API_REQUEST_URL")}/Products/{catalog_id}/Discount",
        headers={
            "Content-Type": "application/json",
            "Accept": "application/json",
            "SecureURL": getenv("SHIFT4SHOP_API_SECURE_URL"),
            "PrivateKey": getenv("SHIFT4SHOP_API_PRIVATE_KEY"),
            "Token": getenv("SHIFT4SHOP_API_TOKEN"),
        },
    )

    if not response.ok:
        return {
            "status": response.status_code,
            "title": "Failed to fetch",
            "message": f"Failed to fetch quantity pricing for Catalog ID {catalog_id}: {response.status_code}",
        }

    return {
        "message": f"Quantity pricing retrieved successfully (ID: {catalog_id})",
        "data": response.json(),
    }

File: src/controllers/quantityPricing.controller.js
export async function getQuantityPricing(req, res, next) {
  const { catalogId } = req.params;
  if (!catalogId) {
    return next({
      status: 400,
      title: "Invalid Catalog ID",
      message: "Invalid Catalog ID provided. Please check the ID and try again.",
    });
  }

  const { API_PRIVATE_KEY, API_REQUEST_URL, API_SECURE_URL, API_TOKEN } = process.env;
  const request = await fetch(`${API_REQUEST_URL}/Products/${catalogId}/Discount`, {
    method: "GET",
    headers: {
      "Content-Type": "application/json",
      Accept: "application/json",
      SecureURL: API_SECURE_URL,
      PrivateKey: API_PRIVATE_KEY,
      Token: API_TOKEN,
    },
  });

  if (!request.ok) {
    return next({
      status: request.status,
      title: "Fetch failed",
      message: `Unable to fetch product quantity pricing: ${response.statusText}`,
    });
  }

  res.status(200).json({
    message: `Quantity pricing retrieved successfully (ID: ${catalogId})`,
    data: await request.json(),
  });
}
